building_sim_plugins/building_ignition_plugins/src/crowd_simulator.cpp

```diff
@@ -87,8 +87,8 @@ void CrowdSimulatorPlugin::PreUpdate(
   std::chrono::duration<double> delta_sim_time_tmp = info.simTime -
     _last_sim_time;
   double delta_sim_time = delta_sim_time_tmp.count();
-  if (_crowd_sim_interface->get_sim_time_step() > delta_sim_time)  // not reach one time sim update
-  {
+  if (_crowd_sim_interface->get_sim_time_step() - delta_sim_time > 1e-6)
+  { // not reach one time sim update
```

Comment:
Curiosity, why is this needed?

Reply:
This controls the menge update sim step. 
The logic is 
```
if (delta_sim_time ==  update_time_step)
{
  delta_sim_time = update_time_step
} else
{
 delta_sim_time = 0.0;
}
```
You can't directly use == to check whether 2 doubles are equal, so I used this expression.
Also, this was to make the code consistent with the gazebo version. Because this expression was adopted in gazebo version.

Reply:
I'm having a closer look at the plugin logic and I'm not sure I understand, why are we calling `_update_all_objects` even when we are not in a simulation time step? From what I see in the function the only thing that is done when we are not at the update time step is [update the external agent](https://github.com/osrf/traffic_editor/blob/ee6a40920d637501974da81d3ab12764b8f78e25/building_sim_plugins/building_ignition_plugins/src/crowd_simulator.cpp#L364-L365), but does updating the external agents pose at a very high frequency (say 1000 Hz) make any difference when the simulator itself only computes steps at 10 Hz?

Reply:
OK, fix the double comparison. Also checked the `delta_time` and `delta_sim_time` logic, and test shows no difference on updating external agent quite at a high frequency (I think initially I thought updating the external agent faster will make the motion planning more accurate) So I deleted the necessary `delta_time` logic, only use `delta_sim_time` at 0.1s to update all objects.